src/Functions/castOrDefault.cpp

```diff
@@ -376,66 +376,475 @@ SELECT accurateCastOrDefault('abc', 'UInt32')
+        toInt32OrDefault_documentation);
+    FunctionDocumentation::Description toInt64OrDefault_description = R"(
+Like [`toInt64`](#toint64), this function converts an input value to a value of type [Int64](../data-types/int-uint.md) but returns the default value in case of an error.
+If no `default` value is passed then `0` is returned in case of an error.
+    )";
+    FunctionDocumentation::Syntax toInt64OrDefault_syntax = "toInt64OrDefault(expr[, default])";
+    FunctionDocumentation::Arguments toInt64OrDefault_arguments = {
+        {"expr", "Expression returning a number or a string representation of a number.", {"String", "Int", "Float"}},
+        {"default", "Optional. The default value to return if parsing is unsuccessful.", {"Int64"}}
+    };
+    FunctionDocumentation::ReturnedValue toInt64OrDefault_returned_value = {"Value of type Int64 if successful, otherwise returns the default value if passed or 0 if not.", {"Int64"}};
+    FunctionDocumentation::Examples toInt64OrDefault_examples = {
+        {"Successful conversion", "SELECT toInt64OrDefault('-64', CAST('-1', 'Int64'))", "-64"},
+        {"Failed conversion", "SELECT toInt64OrDefault('abc', CAST('-1', 'Int64'))", "-1"}
+    };
+    FunctionDocumentation::Category toInt64OrDefault_category = FunctionDocumentation::Category::TypeConversion;
+    FunctionDocumentation::IntroducedIn toInt64OrDefault_introduced_in = {21, 11};
+    FunctionDocumentation toInt64OrDefault_documentation = {toInt64OrDefault_description, toInt64OrDefault_syntax, toInt64OrDefault_arguments, toInt64OrDefault_returned_value, toInt64OrDefault_examples, toInt64OrDefault_introduced_in, toInt64OrDefault_category};
+
     factory.registerFunction("toInt64OrDefault", [](ContextPtr context)
-        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toInt64OrDefault", std::make_shared<DataTypeInt64>()); });
+        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toInt64OrDefault", std::make_shared<DataTypeInt64>()); },
+        toInt64OrDefault_documentation);
+    FunctionDocumentation::Description toInt128OrDefault_description = R"(
+Like [`toInt128`](#toint128), this function converts an input value to a value of type [Int128](../data-types/int-uint.md) but returns the default value in case of an error.
+If no `default` value is passed then `0` is returned in case of an error.
+    )";
+    FunctionDocumentation::Syntax toInt128OrDefault_syntax = "toInt128OrDefault(expr[, default])";
+    FunctionDocumentation::Arguments toInt128OrDefault_arguments = {
+        {"expr", "Expression returning a number or a string representation of a number.", {"String", "Int", "Float"}},
+        {"default", "Optional. The default value to return if parsing is unsuccessful.", {"Int128"}}
+    };
+    FunctionDocumentation::ReturnedValue toInt128OrDefault_returned_value = {"Value of type Int128 if successful, otherwise returns the default value if passed or 0 if not.", {"Int128"}};
```

Comment:
```suggestion
    FunctionDocumentation::ReturnedValue toInt128OrDefault_returned_value = {"Returns a value of type Int128 if successful, otherwise returns the default value if passed, or 0 if not.", {"Int128"}};
```